theano/tensor/nnet/abstract_conv.py

```diff
@@ -177,6 +177,187 @@ def conv2d(input,
     return conv_op(input, filters)
 
 
+def deconv(input,
+           filters,
+           input_shape=None,
+           output_shape=None,
+           filter_shape=None,
+           border_mode='valid',
+           subsample=(1, 1),
+           filter_flip=True):
+    """This function will build the symbolic graph for deconvoloving a
+    mini-batch of a stack of 2D inputs with a set of 2D filters (using
+    gradient of convolution), such that it upsamples the input to the
+    desired output resolution.
```

Comment:
OK, so maybe we should name the functions `conv2d_grad_inputs` and `conv2d_grad_weights`, and mention in the docstring of `conv2d_grad_inputs` that this is what Zeiler et al. call "deconvolution".
I think it is important to keep the "conv2d" part in the name, and we do not use `camelCase` in function names.